pkg/api/types.go

```diff
@@ -1815,7 +1815,8 @@ type Secret struct {
 	TypeMeta   `json:",inline"`
 	ObjectMeta `json:"metadata,omitempty"`
 
-	// Data contains the secret data.  Each key must be a valid DNS_SUBDOMAIN.
+	// Data contains the secret data.  Each key must be a valid DNS_SUBDOMAIN
```

Comment:
> If we intend for these to be used for file names, why not just say "valid POSIX file name (excluding '.' and '..')"

Can we relax this one aspect before we close on https://github.com/GoogleCloudPlatform/kubernetes/issues/8236?